traits/traits.py

```diff
@@ -207,6 +207,75 @@ def date_editor ( ):
 
     return DateEditor
 
+def _is_hastraits_instance(handler):
+    """ Does a trait handler or type expect a HasTraits subclass instance?
+    """
+    from traits.api import HasTraits, BaseInstance, TraitInstance
+
+    if isinstance(handler, TraitInstance):
+        cls = handler.aClass
+    elif isinstance(handler, BaseInstance):
+        cls = handler.klass
+    else:
+        return False
+    return issubclass(cls, HasTraits)
+
+def _is_string_trait(handler):
+    """ Is this one of the string matching trait handlers
+    """
+    from traits.api import BaseStr, String, TraitString
+    return isinstance(handler, (BaseStr, String, TraitString))
+
+def _instance_handler_factory(handler):
+    """ Get the instance factory of an Instance or TraitInstance
+    """
+    from traits.api import BaseInstance, TraitInstance
+
+    if isinstance(handler, TraitInstance):
+        return handler.aClass
+    elif isinstance(handler, BaseInstance):
+        return handler.default_value
+
+def list_editor(trait, handler):
+    """ Factory that constructs an appropriate editor for a list.
+    """
+    item_handler = handler.item_trait.handler
+    if _is_hastraits_instance(item_handler):
+        from traitsui.table_column import ObjectColumn
+        from traitsui.table_filter import (EvalFilterTemplate,
+            RuleFilterTemplate, MenuFilterTemplate, EvalTableFilter)
+        from traitsui.api import TableEditor
+
+        return TableEditor(
+            filters=[RuleFilterTemplate, MenuFilterTemplate,
+                     EvalFilterTemplate],
+            edit_view='',
+            orientation='vertical',
+            search=EvalTableFilter(),
+            deletable=True,
+            show_toolbar=True,
+            reorderable=True,
+            row_factory=_instance_handler_factory(item_handler)
+        )
+    elif _is_string_trait(item_handler):
+        from traitsui.api import ListStrEditor
+
+        return ListStrEditor(
+            auto_add=True,
+            title=trait.title if trait.title else '',
+            title_name=trait.title_name if trait.title_name else '',
+        )
+    else:
+        from traitsui.api import ListEditor
+
+        return ListEditor(
+            trait_handler=handler,
+            rows=trait.rows if trait.rows else 5,
+            use_notebook=bool(trait.use_notebook),
```

Comment:
I don't think we need or want the `bool` call here.

Reply:
Yes we do. `ListEditor.use_notebook` is a `Bool` trait, but the default value of `trait.use_notebook` is `None`.

Or would you prefer `trait.use_notebook is True`? :)

Reply:
Ah. :-(  No, I don't prefer `trait.use_notebook is True`. I take back my objection.

Reply:
Thinking about this a little more, `trait.use_notebook if trait.use_notebook is not None else False` may be better than `bool` since that will raise on truthy or falsy values that aren't `True` or `False`

Reply:
:+1: